### PR TITLE
`arrow2-convert` migration 5: sunset `arrow2-convert`

### PR DESCRIPTION
**Pull request by teh-cmc**
The end of our wonderful journey.

- `NumInstances` control column now has an actual dedicated component type.
- `EntityPath` is now a component.
- `Into<Cow<Self>>` impls have been cleaned up to generate way less code.
- `arrow2_convert` is fully gone.

---

`arrow2-convert` migration PR series:
- #3853 
- #3855
- #3897
- #3902
- #3917

---

- DNM: requires #3902 first

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3917) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/3917)
- [Docs preview](https://rerun.io/preview/3d6bae9918242bb6ae216fa397c3d86ff9b65816/docs) 
- [Examples preview](https://rerun.io/preview/3d6bae9918242bb6ae216fa397c3d86ff9b65816/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_types/src/components/radius.rs [hunk 1]
```diff
@@ -34,20 +34,7 @@ impl From<Radius> for f32 {
...
-    }
-}
-
-impl<'a> From<&'a Radius> for ::std::borrow::Cow<'a, Radius> {
-    #[inline]
-    fn from(value: &'a Radius) -> Self {
-        std::borrow::Cow::Borrowed(value)
-    }
-}
+::re_types_core::macros::impl_into_cow!(Radius);
```
Comment (teh-cmc) [inline crates/re_types/src/components/radius.rs:37]: missing a linefeed here and everywhere

File: rerun_cpp/src/rerun/datatypes/u_int32.cpp
Comment (teh-cmc) [inline rerun_cpp/src/rerun/datatypes/u_int32.cpp:1]: this file and its hpp pear dont't exist, remove

File: crates/re_arrow_store/src/store_gc.rs [hunk 2]
```diff
@@ -373,7 +373,7 @@ impl DataStore {
         self.timeless_tables.retain(|_, table| {
             // If any column is non-empty, we need to keep this table
             for num in &table.col_num_instances {
-                if num != &0 {
+                if num != &0.into() {
```
Comment (emilk) [inline crates/re_arrow_store/src/store_gc.rs:376]: this reads weird

File: crates/re_log_types/src/num_instances.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+impl From<NumInstances> for u32 {
+    fn from(val: NumInstances) -> Self {
+        val.0
+    }
+}
+
+impl From<u32> for NumInstances {
+    fn from(value: u32) -> Self {
```
Comment (emilk) [inline crates/re_log_types/src/num_instances.rs:26]:
```suggestion
    #[inline]
    fn from(val: NumInstances) -> Self {
        val.0
    }
}

impl From<u32> for NumInstances {
    #[inline]
    fn from(value: u32) -> Self {
```

File: crates/re_types/definitions/rerun/datatypes/uint32.fbs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+include "arrow/attributes.fbs";
+include "python/attributes.fbs";
+include "fbs/attributes.fbs";
+include "rust/attributes.fbs";
+
+namespace rerun.datatypes;
+
+/// A 32bit integer.
```
Comment (emilk) [inline crates/re_types/definitions/rerun/datatypes/uint32.fbs:8]:
```suggestion
/// A 32bit unsigned integer.
```

File: crates/re_types_builder/src/codegen/rust/serializer.rs [hunk 5]
```diff
@@ -574,7 +574,7 @@ fn quote_arrow_field_serializer(
                                     })
                                     .transpose()
                                 })
-                                .collect::<::re_types_core::SerializationResult<Vec<_>>>()?
```
Comment (emilk) [inline crates/re_types_builder/src/codegen/rust/serializer.rs:577]: a `;` after a `collect` that has no `let` binding? what is going on here?

File: rerun_cpp/src/rerun/datatypes/u_int32.cpp [hunk 6]
```diff
@@ -0,0 +1,47 @@
+// DO NOT EDIT! This file was auto-generated by crates/re_types_builder/src/codegen/cpp/mod.rs
```
Comment (emilk) [inline rerun_cpp/src/rerun/datatypes/u_int32.cpp:1]: weird file-name `u_int32.cpp`

accidentally comited old file?
why wasn't this pruned by `remove_orphaned_files`?

File: rerun_cpp/src/rerun/datatypes/u_int32.cpp
Comment (emilk) [reply rerun_cpp/src/rerun/datatypes/u_int32.cpp:1]: It should have been caught by `remove_orphaned_files`

**Review by emilk — APPROVED** [review]
good riddance!

File: crates/re_types_builder/src/codegen/rust/serializer.rs (hunk 5 above)
Comment (teh-cmc) [reply crates/re_types_builder/src/codegen/rust/serializer.rs]: Yeah the whole "who's responsible for the trailing ';'" is a mess in general.

I'll clean that up during the switch to arrow1, likely by introducing a new rule that if your quote block does not do the assignment, then it should not include the trailing comma.

**Pull request merged**